sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1008,6 +1008,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val HIVE_METASTORE_PARTITION_PRUNING_EVAL_CLIENT_SIDE =
+    buildConf("spark.sql.hive.metastorePartitionPruningEvalClientSide")
```

Comment:
This name is confusing. We can probably make it general: `spark.sql.hive.partitionPruningFastFallback`

Reply:
Modified to `spark.sql.hive.metastorePartitionPruningFastFallback`